### PR TITLE
Jhakulin/azure ai client agents

### PR DESCRIPTION
**Pull request by jhakulin**
Agents TypeSpec definitions

### CODE REVIEW

File: specification/ai/Azure.AI.Client/files/models.tsp [hunk 1]
```diff
@@ -0,0 +1,129 @@
+import "@typespec/versioning";
```
Comment (dargilco) [inline specification/ai/Azure.AI.Client/files/models.tsp:1]: Can you please put everything related to Agents under the  
specification/ai/Azure.AI.Client/agents folder? You can have multiple subfolders there, like:
/agents/agents/
/agents/files/
/agents/messages/
...

File: specification/ai/OpenAI.Assistants/main.tsp [hunk 2]
```diff
@@ -47,8 +47,8 @@ namespace Azure.AI.OpenAI {
   }
 )
 @versioned(ServiceApiVersions)
-@doc("Azure OpenAI APIs for Assistants.")
-namespace Azure.AI.OpenAI.Assistants {
+@doc("Azure APIs for Agents.")
+namespace Azure.AI.Client.Agents {
   @doc("The known set of supported API versions.")
```
Comment (dargilco) [inline specification/ai/OpenAI.Assistants/main.tsp:52]: Why changes in OpenAI file?

Comment (jhakulin) [reply specification/ai/OpenAI.Assistants/main.tsp:52]: accidental, will remove it

File: specification/ai/Azure.AI.Client/files/models.tsp (hunk 1 above)
Comment (jhakulin) [reply specification/ai/Azure.AI.Client/files/models.tsp:1]: Why?

File: specification/ai/Azure.AI.Client/agents/main.tsp [hunk 3]
```diff
@@ -0,0 +1,2 @@
+import "./models.tsp";
+import "./routes.tsp";
```
Comment (dargilco) [inline specification/ai/Azure.AI.Client/agents/main.tsp:2]: No need for this... we should have only one "main.tsp" in the root folder of project, that imports all that is needed.

Comment (jhakulin) [reply specification/ai/Azure.AI.Client/agents/main.tsp:2]: It follows what we have in OpenAI Assistants. If we restructure it, the comparisons will be harder in future.

File: specification/ai/Azure.AI.Client/vector_stores/common/main.tsp [hunk 4]
```diff
@@ -0,0 +1,89 @@
+import "@typespec/http";
```
[the remark this answers is not in the document]
Comment (jhakulin) [reply specification/ai/Azure.AI.Client/vector_stores/common/main.tsp:1]: Again it follows the OpenAI Assistants

File: specification/ai/Azure.AI.Client/vector_stores/file_batches/models.tsp [hunk 5]
```diff
@@ -0,0 +1,45 @@
+namespace Azure.AI.Client.Agents;
```
Comment (dargilco) [inline specification/ai/Azure.AI.Client/vector_stores/file_batches/models.tsp:1]: Same here... looks like all these should be ain a models.tsp file.

Comment (jhakulin) [reply specification/ai/Azure.AI.Client/vector_stores/file_batches/models.tsp:1]: Same story

File: specification/ai/Azure.AI.Client/files/models.tsp (hunk 1 above)
Comment (jhakulin) [reply specification/ai/Azure.AI.Client/files/models.tsp:1]: I have put the files like this now to follow OpenAI assistants structure as close as possible

File: specification/ai/OpenAI.Assistants/main.tsp (hunk 2 above)
Comment (jhakulin) [reply specification/ai/OpenAI.Assistants/main.tsp]: removed the accidental change

File: specification/ai/Azure.AI.Client/vector_stores/file_batches/models.tsp (hunk 5 above)
Comment (jhakulin) [reply specification/ai/Azure.AI.Client/vector_stores/file_batches/models.tsp]: restructured

File: specification/ai/Azure.AI.Client/agents/main.tsp (hunk 3 above)
Comment (jhakulin) [reply specification/ai/Azure.AI.Client/agents/main.tsp]: restructured

**Review by dargilco — APPROVED** [review]

**Pull request merged**